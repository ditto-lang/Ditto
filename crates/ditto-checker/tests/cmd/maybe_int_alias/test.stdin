module Test exports (..)

import Maybe

type alias Maybe = Maybe.Maybe(Int)

just_int = fn (i: Int): Maybe -> Maybe.Just(i)
