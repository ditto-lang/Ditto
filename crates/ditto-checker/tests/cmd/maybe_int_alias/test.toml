bin.name = "ditto-checker-testbin-check-module"
args = ["maybe_int_alias.ditto"]
fs.sandbox = true
