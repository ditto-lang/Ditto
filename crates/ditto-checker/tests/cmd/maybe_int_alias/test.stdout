{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {
      "Maybe": {
        "Alias": {
          "doc_comments": [],
          "doc_position": 0,
          "kind": "Type",
          "aliased_type": {
            "type": "Call",
            "data": {
              "function": {
                "type": "Constructor",
                "data": {
                  "constructor_kind": {
                    "Function": {
                      "parameters": [
                        "Type"
                      ]
                    }
                  },
                  "canonical_value": {
                    "module_name": [
                      null,
                      [
                        "Maybe"
                      ]
                    ],
                    "value": "Maybe"
                  },
                  "source_value": {
                    "module_name": "Maybe",
                    "value": "Maybe"
                  }
                }
              },
              "arguments": [
                {
                  "type": "PrimConstructor",
                  "data": "Int"
                }
              ]
            }
          },
          "alias_variables": []
        }
      }
    },
    "constructors": {},
    "values": {
      "just_int": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "Function",
          "data": {
            "parameters": [
              {
                "type": "PrimConstructor",
                "data": "Int"
              }
            ],
            "return_type": {
              "type": "Call",
              "data": {
                "function": {
                  "type": "Constructor",
                  "data": {
                    "constructor_kind": {
                      "Function": {
                        "parameters": [
                          "Type"
                        ]
                      }
                    },
                    "canonical_value": {
                      "module_name": [
                        null,
                        [
                          "Maybe"
                        ]
                      ],
                      "value": "Maybe"
                    },
                    "source_value": {
                      "module_name": null,
                      "value": "Maybe"
                    }
                  }
                },
                "arguments": [
                  {
                    "type": "PrimConstructor",
                    "data": "Int"
                  }
                ]
              }
            }
          }
        }
      }
    }
  },
  "types": {
    "Maybe": {
      "Alias": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 53,
          "end_offset": 58
        },
        "kind": "Type",
        "aliased_type": {
          "type": "Call",
          "data": {
            "function": {
              "type": "Constructor",
              "data": {
                "constructor_kind": {
                  "Function": {
                    "parameters": [
                      "Type"
                    ]
                  }
                },
                "canonical_value": {
                  "module_name": [
                    null,
                    [
                      "Maybe"
                    ]
                  ],
                  "value": "Maybe"
                },
                "source_value": {
                  "module_name": "Maybe",
                  "value": "Maybe"
                }
              }
            },
            "arguments": [
              {
                "type": "PrimConstructor",
                "data": "Int"
              }
            ]
          }
        },
        "alias_variables": []
      }
    }
  },
  "constructors": {},
  "values": {
    "just_int": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 80,
        "end_offset": 88
      },
      "expression": {
        "expression": "Function",
        "data": {
          "span": {
            "start_offset": 91,
            "end_offset": 126
          },
          "function_type": {
            "type": "Function",
            "data": {
              "parameters": [
                {
                  "type": "PrimConstructor",
                  "data": "Int"
                }
              ],
              "return_type": {
                "type": "Call",
                "data": {
                  "function": {
                    "type": "Constructor",
                    "data": {
                      "constructor_kind": {
                        "Function": {
                          "parameters": [
                            "Type"
                          ]
                        }
                      },
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Maybe"
                          ]
                        ],
                        "value": "Maybe"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "Maybe"
                      }
                    }
                  },
                  "arguments": [
                    {
                      "type": "PrimConstructor",
                      "data": "Int"
                    }
                  ]
                }
              }
            }
          },
          "binders": [
            [
              {
                "Variable": {
                  "span": {
                    "start_offset": 95,
                    "end_offset": 96
                  },
                  "name": "i"
                }
              },
              {
                "type": "PrimConstructor",
                "data": "Int"
              }
            ]
          ],
          "body": {
            "expression": "Call",
            "data": {
              "span": {
                "start_offset": 113,
                "end_offset": 126
              },
              "call_type": {
                "type": "Call",
                "data": {
                  "function": {
                    "type": "Constructor",
                    "data": {
                      "constructor_kind": {
                        "Function": {
                          "parameters": [
                            "Type"
                          ]
                        }
                      },
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Maybe"
                          ]
                        ],
                        "value": "Maybe"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "Maybe"
                      }
                    }
                  },
                  "arguments": [
                    {
                      "type": "PrimConstructor",
                      "data": "Int"
                    }
                  ]
                }
              },
              "function": {
                "expression": "ImportedConstructor",
                "data": {
                  "span": {
                    "start_offset": 113,
                    "end_offset": 123
                  },
                  "constructor_type": {
                    "type": "Function",
                    "data": {
                      "parameters": [
                        {
                          "type": "PrimConstructor",
                          "data": "Int"
                        }
                      ],
                      "return_type": {
                        "type": "Call",
                        "data": {
                          "function": {
                            "type": "Constructor",
                            "data": {
                              "constructor_kind": {
                                "Function": {
                                  "parameters": [
                                    "Type"
                                  ]
                                }
                              },
                              "canonical_value": {
                                "module_name": [
                                  null,
                                  [
                                    "Maybe"
                                  ]
                                ],
                                "value": "Maybe"
                              },
                              "source_value": {
                                "module_name": null,
                                "value": "Maybe"
                              }
                            }
                          },
                          "arguments": [
                            {
                              "type": "PrimConstructor",
                              "data": "Int"
                            }
                          ]
                        }
                      }
                    }
                  },
                  "constructor": {
                    "module_name": [
                      null,
                      [
                        "Maybe"
                      ]
                    ],
                    "value": "Just"
                  }
                }
              },
              "arguments": [
                {
                  "Expression": {
                    "expression": "LocalVariable",
                    "data": {
                      "span": {
                        "start_offset": 124,
                        "end_offset": 125
                      },
                      "variable_type": {
                        "type": "PrimConstructor",
                        "data": "Int"
                      },
                      "variable": "i"
                    }
                  }
                }
              ]
            }
          }
        }
      }
    }
  },
  "values_toposort": [
    "just_int"
  ]
}